### DISCUSSION TITLE
Mount Points & SaaS Boost

### DISCUSSION THREAD

**Discussion by AffiTheCreator** [awslabs/aws-saas-boost#37] [Q&A] [answered]
I have an issue with mount point volumes and SaaS boost, it's more of a user problem rather than a code one.  I was hoping someone could help me. I have read all the documentation about mounted volumes in docker and AWS. But still can't get it to work within AWS. 

## What I have!

1. I have an app that is divided into 2 parts, the main software and modules, they work as an add-on.
2. I'm trying to have the add-ons folder mounted in an EFS with the File System capability of SaaS boost
3. I have modified the main part of the software to accept the new directory. 
4. I have generated Deb files to install both parts, the main part and each individual add-on, they are located inside /Modules
5. When building the container the dockerfile runs a script that installs all the deb files correctly.
6. If I use --mount flag like so ` docker run --name foo --publish 8000:80 --mount source=module_vol,target=/foo bar ` it works, but the volume is created inside the default folder and I cannot change that, I think, also this type of volumes are not compatible with fargate only EC2. I'm running fargate, so I can't use it to test for AWS.
7. if I use the -v f flag ` docker run --name SaasBoost --publish 8000:80 -v E:\dockerVolume\aws:/foo bar ` it also works,  with the small catch that, all the files that have been previously installed during the docker build process have disappeared. My software then creates the folder structure but it's empty.
8. Since Fargate only supports bind volumes I can only use -v flag
9. I have uploaded a container to AWS ECR Repo and I have tried launching it with the mounted volume and without. If I use internal storage, meaning no mount points my software works as intended but as soon as I introduced the mount points it starts complaining about missing directories, more specifically the mounted volumes.
10. start.sh launches a series of installations. I have tested the installation script in previous tenants and always worked.
    - One of the installation files creates the foo directory and gives it proper read-write access so the folder structure exists
    
Here is part of my dockerfile:
```
#Pull base image
FROM ubuntu:18.04

.
.
. Not important stuff
.
.
RUN start.sh
# At this point all the necessary files have been copied to the correct folders , the documentation states 
# any files inside the directory will be copied to the mounted volume. 
VOLUME ["/foo"]

# I try again installing the modules files to foo directory with no effect
WORKDIR /Modules
RUN apt-get install ./modules.*.deb

WORKDIR /
ENTRYPOINT service main_software start && tail -f /dev/null
```

## The problem
I need /foo directory not to be empty when I use the -v flag , and the documentation states any file inside the container's folder will be copied to the bind mount volume. This is not happening.

**Comment by hariohmprasath** [comment, marked as answer]
Hi @AffiTheCreator,

There is an issue with the code that prevents EFS volume from getting mounted in the main container. I have pushed the changes part of this commit which should resolve this issue

[https://github.com/awslabs/aws-saas-boost/pull/33/commits/76c9d957b4d04b96aee1ee0328697b50125c7cce](https://github.com/awslabs/aws-saas-boost/pull/33/commits/76c9d957b4d04b96aee1ee0328697b50125c7cce)

Once the changes are merged you don't need to install any drivers for mounting EFS, its by default mounted on the `Mount Point` specified while creating the app (in my case its `/mnt`). 

![Screen Shot 2021-05-18 at 5 52 45 PM](https://user-images.githubusercontent.com/2615325/118740995-1a8fa900-b802-11eb-919f-fce0a90d60be.png)


You can directly write and read from these volume using your program. Hope this answers your question.

~ Hari

**Comment by 4patelr** [comment]
The fixes by Hari have been merged into the main branch.
